### PR TITLE
OpenMP support for TRD tracking

### PR DESCRIPTION
**Pull request by martenole**
Hi David, I have some changes to the TRD tracking that I would like to push, but there are conflicts with your master branch. Can we still merge them?
OpenMP works fine now. I still have to check the transformation matrices from the new flat geometry.
Cheers,
Ole

### CODE REVIEW

File: TPCCAGPUTracking/CMakeLists.txt [hunk 1]
```diff
@@ -196,6 +196,10 @@ if(ALITPCCOMMON_BUILD_TYPE STREQUAL "ALIROOT")
     # Additional compilation flags
     set_target_properties(${MODULE} PROPERTIES COMPILE_FLAGS "")
 
+    # OpenMP support
+    find_package(OpenMP)
+    target_link_libraries(${MODULE} OpenMP::OpenMP_CXX)
```
Comment (davidrohr) [inline TPCCAGPUTracking/CMakeLists.txt:201]: we cannot change the AliRoot cmake here, you can have this change only locally, but not push it.
In order not to break AliRoot, please also add an
`#ifdef HLTCA_HAVE_OPENMP`
before all omp pragmas and omp_ API calls.

File: TPCCAGPUTracking/GlobalTracker/AliGPUReconstruction.cxx [hunk 2]
```diff
@@ -393,6 +393,20 @@ int AliGPUReconstruction::RunTRDTracking()
 	{
 		const AliHLTTPCGMMergedTrack& trk = mIOPtrs.mergedTracks[i];
 		if (!trk.OK()) continue;
+		if (trk.Looper()) continue;
```
Comment (davidrohr) [inline TPCCAGPUTracking/GlobalTracker/AliGPUReconstruction.cxx:396]: OK for now, should be cleaned up later in the TPC code.

File: TPCCAGPUTracking/GlobalTracker/AliGPUReconstruction.h [hunk 3]
```diff
@@ -138,9 +138,17 @@ class AliGPUReconstruction
 	{
 		return getPointerWithAlignment<S>(reinterpret_cast<size_t&>(basePtr), nEntries);
 	}
-	template <class T, class S> static inline void computePointerWithAlignment(T*& basePtr, S*& objPtr, size_t nEntries = 1)
+
+	template <class T, class S> static inline void computePointerWithAlignment(T*& basePtr, S*& objPtr, size_t nEntries = 1, bool runConstructor = false)
 	{
 		objPtr = getPointerWithAlignment<S>(reinterpret_cast<size_t&>(basePtr), nEntries);
+    if (runConstructor)
```
Comment (davidrohr) [inline TPCCAGPUTracking/GlobalTracker/AliGPUReconstruction.h:145]: please fix the indentation

File: TPCCAGPUTracking/TRDTracking/AliHLTTRDTracker.cxx [hunk 4]
```diff
@@ -74,94 +110,57 @@ AliHLTTRDTracker::~AliHLTTRDTracker()
...
   if (fIsInitialized) {
-    delete[] fTracklets;
-    delete[] fTracks;
-    delete[] fSpacePoints;
-    delete[] fHypothesis;
-    delete[] fCandidates;
-    delete[] fNtrackletsInChamber;
-    delete[] fTrackletIndexArray;
-    delete[] fR;
-    if (!fExternalGeometry) delete fGeo;
```
Comment (davidrohr) [inline TPCCAGPUTracking/TRDTracking/AliHLTTRDTracker.cxx:85]: I think we have to keep the destructor for debug at keast

File: TPCCAGPUTracking/TRDTracking/AliHLTTRDTrackerComponent.cxx [hunk 5]
```diff
@@ -201,14 +204,18 @@ int AliHLTTRDTrackerComponent::DoInit( int argc, const char** argv ) {
 
   iResult = ReadConfigurationString( arguments.Data() );
 
+  fGeo = new AliHLTTRDGeometry();
```
Comment (davidrohr) [inline TPCCAGPUTracking/TRDTracking/AliHLTTRDTrackerComponent.cxx:207]: You could put the other checks here, if the AliGeometry for TRD is actually available, which were with the Geometry initialization before.

**Review by davidrohr — CHANGES_REQUESTED** [review]
Hi @martenole : I think in general it is fine, I left some minor comments inline.
But please do not use git merge, but rebase your branch onto the master for the pull request.

**Comment by martenole** [comment]
I have applied the requested changes and will open another pull request with the current code.

**Pull request closed without merging**